File: JavaToKotlin/app/src/main/java/io/dustin/apps/board/api/bookmark/BookmarkController.java
package io.dustin.apps.board.api.bookmark;

import io.dustin.apps.board.api.usecase.bookmark.DeleteBookmarkUseCase;
import io.dustin.apps.board.api.usecase.bookmark.WriteBookmarkUseCase;
import io.dustin.apps.board.domain.bookmark.model.dto.BookMarkDto;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1")
@RequiredArgsConstructor
public class BookmarkController {

    private final WriteBookmarkUseCase writeBookmarkUseCase;
    private final DeleteBookmarkUseCase deleteBookmarkUseCase;

    @PostMapping("/bookmarks/add")
    public BookMarkDto add(@RequestBody BookMarkDto bookMarkDto) {
        System.out.println("북마크 저장했습니다.");
        return writeBookmarkUseCase.execute(bookMarkDto.userId(), bookMarkDto.boardId());
    }

    @DeleteMapping("/bookmarks/cancel")
    public BookMarkDto cancel(@RequestBody BookMarkDto bookMarkDto) {
        System.out.println("북마크 삭제했습니다.");
        return deleteBookmarkUseCase.execute(bookMarkDto.userId(), bookMarkDto.boardId());
    }

}

File: apps/src/main/kotlin/io/dustin/apps/board/api/bookmark/BookmarkController.kt
package io.dustin.apps.board.api.bookmark

import io.dustin.apps.board.api.bookmark.request.command.BookmarkCreateCommand
import io.dustin.apps.board.api.bookmark.request.command.BookmarkDeleteCommand
import io.dustin.apps.board.api.usecase.bookmark.DeleteBookmarkUseCase
import io.dustin.apps.board.api.usecase.bookmark.WriteBookmarkUseCase
import io.dustin.apps.board.domain.bookmark.model.dto.BookMarkDto
import io.dustin.apps.common.code.CommonMessage
import io.dustin.apps.common.model.response.CommonResponse
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.time.LocalDateTime

@RestController
@RequestMapping("/api/v1/bookmarks")
@Tag(name = "북마크 관련 API", description = "북마크 관련 API를 제공한다.")
class BookmarkController (

    private val writeBookmarkUseCase: WriteBookmarkUseCase,
    private val deleteBookmarkUseCase: DeleteBookmarkUseCase

) {

    @ResponseStatus(HttpStatus.OK)
    @PostMapping("/add")
    @Operation(
        summary = "북마크 추가",
        description = "북마크 추가"
    )
    fun add(@RequestBody @Valid command: BookmarkCreateCommand): CommonResponse {
        writeBookmarkUseCase.execute(command.userId, command.boardId)
        return CommonResponse(
            code = HttpStatus.OK.value(),
            message = CommonMessage.SUCCESS.code,
            timestamp = LocalDateTime.now()
        )
    }

    @ResponseStatus(HttpStatus.OK)
    @DeleteMapping("/cancel")
    @Operation(
        summary = "북마크 해제",
        description = "북마크 해제"
    )
    fun cancel(@RequestBody @Valid command: BookmarkDeleteCommand): CommonResponse {
        deleteBookmarkUseCase.execute(command.userId, command.boardId)
        return CommonResponse(
            code = HttpStatus.OK.value(),
            message = CommonMessage.SUCCESS.code,
            timestamp = LocalDateTime.now()
        )
    }
}





